### PR TITLE
OAuth 連携の不具合修正

### PR DESCRIPTION
**Pull request by hiroaki**
OAuth 連携時に、すでに他のユーザに連携されていた場合には Authorization モデルのほうでバリデーションエラーとなるため、データは更新されないのですが、画面の flash は連携が成功したと表示されていました。

その修正でロジックを見直していた中で、ユーザが同一のproviderの複数のレコードを持ててしまっていることに気がつき、その修正も加えました。

また念の為ログを多めに書き出すようにしています。

以下は Copilot によるサマリーです：

This pull request enhances the handling of user authorization linking in the OmniAuth callbacks and improves test coverage. The key changes include stricter validation rules for `Authorization` records, improved logging for potential conflicts, and expanded test cases to cover edge scenarios.

### Enhancements to Authorization Logic:
* Updated `link_provider_or_alert` in `OmniAuthCallbacksController` to handle cases where a provider/UID pair is already linked to another user, adding warnings and detailed logging for conflicts.
* Added a new validation to enforce the uniqueness of the `provider` scoped to `user_id` in the `Authorization` model.
* Introduced a new class method `provider_uid_exists?` in the `Authorization` model to check for existing provider/UID pairs.

### Improvements to Test Coverage:
* Expanded `Authorization` model tests to validate new constraints, including duplicate provider/UID pairs across users and duplicate user/provider pairs.
* Enhanced request specs for OmniAuth callbacks to verify behavior in edge cases, such as self-linking, linking with a provider/UID already linked to another user, and handling race conditions during linking. [[1]](diffhunk://#diff-b175cb520b316cea2754c90273d0fa0fc9f18424afeea7a370fa2b734be2b88dL31-R91) [[2]](diffhunk://#diff-b175cb520b316cea2754c90273d0fa0fc9f18424afeea7a370fa2b734be2b88dL74-R117)

### Test Cleanup and Refactoring:
* Refactored existing test descriptions and structure for clarity and consistency, replacing Japanese comments with English for broader accessibility. [[1]](diffhunk://#diff-b7c13b23d47e4f2f7061baeea9d0ac76ab8bd3176cc0c170a2d9b850266a6c20L4-R56) [[2]](diffhunk://#diff-b175cb520b316cea2754c90273d0fa0fc9f18424afeea7a370fa2b734be2b88dL87-R136)

### CODE REVIEW

File: app/controllers/users/omniauth_callbacks_controller.rb [hunk 1]
```diff
@@ -31,11 +31,26 @@ def failure
   def link_provider_or_alert(auth, user, locale)
     provider_name = provider_display_name(auth)
 
-    if user.linked_with?(auth.provider) && user.provider_uid(auth.provider) != auth.uid
-      flash[:alert] = I18n.t("devise.omniauth_callbacks.provider.already_linked", provider: provider_name)
+    if Authorization.provider_uid_exists?(auth.provider, auth.uid)
+      if user.provider_uid(auth.provider) == auth.uid
+        # 正常：自身のuserでリンクされている
```
[the remark this answers is not in the document]
Comment (hiroaki) [reply app/controllers/users/omniauth_callbacks_controller.rb:36]: コメント文については気にしないでください。
